.github/scripts/check_content.py

```diff
@@ -0,0 +1,57 @@
+import os.path
+from os import walk
+
+
+def check_content() -> int:
+    """
+    Walks through the filetree rooted at the current working directory.
+    For each file that it finds, it validates the file
+    """
+    error_count = 0
+    DIRECTORIES_TO_EXCLUDE = ['.git', '.github', '.idea', 'venv', 'DO NOT COMMIT']
+    FILES_TO_EXCLUDE = ['.DS_Store', '.gitignore']
+
+    for root, dirs, files in walk('.', topdown=True):
+        dirs[:] = [d for d in dirs if d not in DIRECTORIES_TO_EXCLUDE]
+        files[:] = [f for f in files if f not in FILES_TO_EXCLUDE]
+        for file in files:
+            relative_path = os.path.join(root, file)
+            error_count += check_file(relative_path, file)
+    return error_count
+
+
+def check_file(relative_path: str, file: str) -> int:
+    """
+    Check the given file for any issues.
+
+    Writes error messages to the console, and returns the number of errors found
+
+    :param relative_path: The path of the file, including the directory, relative to the vault's root
+    :param file: The name of the file, without any directory
+    :return: The number of errors found in the given file 
+    """
+    errors = 0
+
+    if '.' not in file:
+        print(f'Error:\n  This file has no extension: consider adding ".md" to its name:\n  {relative_path} ')
+        errors += 1
+
+    # Other checks may be added here in future
+    return errors
+
+
+def get_root_of_vault() -> str:
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    up1 = os.path.dirname(dir_path)
+    up2 = os.path.dirname(up1)
+    return up2
+
+
+def main() -> int:
```

Comment:
Docstring?

Reply:
I see what you mean...

On reflection, I think at the moment I'd rather leave the code as the documentation - as this seems like the sort of thing that might become out of date if it's duplicated.

But that prompted me to make these two renaming, so the function names themselves are at least self-documenting:

- `main()` -> `check_content_of_vault()
- `check_content()` -> `check_content_of_working_directory()`

Thanks for the prod!

Reply:
Maybe I'm misunderstanding the kind of docs you meant?

Reply:
Yes: a docstring is this.
```python
def something():
   """
   This is a docstring explaining what the method does.
   """
```

Reply:
> Maybe I'm misunderstanding the kind of docs you meant?

Hah - I was totally ambiguous - I was meaning what sort of information was worth documenting, that wasn't already clear from the code itself.

But I will write some general overall statement about the types of checks being run, like that it's checking the vault info/contents, rather than the infrastructure.

Reply:
I've added a `--help` arg to `check_content.py`

```
python3 ./check_content.py --help
usage: check_content.py [-h]

Check for issues with the content of the Hub vault (such as errors in file names).

optional arguments:
  -h, --help  show this help message and exit
```

And added a comment at the top of the file:

```python
# This code is for validating the **Content** of the Hub, such as file names,
# as opposed to validating the Python code and other infrastructure.
```

Reply:
Thanks @dbarenholz - see above for the changes I made. Resolving conversation now.